qiskit_ibm_runtime/utils/json.py

```diff
@@ -205,6 +206,13 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
                 serializer=lambda buff, data: qpy_serialization.dump(data, buff),
             )
             return {"__type__": "QuantumCircuit", "__value__": value}
+        if isinstance(obj, Parameter):
+            value = _serialize_and_encode(
+                data=obj,
+                serializer=qpy_serialization._write_parameter,
```

Comment:
This won't work starting in 0.20.0 as this private method has moved on `main`, we added aliases for the other private functions the provider was using previously to avoid a hard breakage, but this wasn't in the list and I don't think we want to add more use of private methods here. This raises https://github.com/Qiskit/qiskit-ibm-runtime/issues/147#issuecomment-1040736485 again and that's how we should be fixing this.

Reply:
Thanks Matthew! I will merge this PR right now to unblock testing primitives and then immediately start working on copying qpy module so we can have a copy here before 0.20.0.